### DISCUSSION TITLE
An official basic template

### DISCUSSION THREAD

**Discussion by AJ1000** [havit/Havit.Blazor#934] [General]
I made a basic web app 8 solution with the most basic set up here

https://github.com/coderdnewbie/BlazorHavitDemo

I think it would be useful for something similar to be in the official Havit repo, so others will be able to see how the dotnet templates work with Havit. https://github.com/havit/NewProjectTemplate-Blazor seems too complex for someone just getting introduced to Havit.

What do you think?

**Comment by hakenr** [comment]
We're currently working on a major _Getting Started_ overhaul.
I believe the new instructions should cover beginners.
Stay tuned - it should be published tomorrow.
We can continue discussing the options then.

**Comment by coderdnewbie** [comment]
Thanks. It would be cool if you replaced the default sidebar with the Havit Sidebar as well in the new instructions as the Havit Sidebar is better.

**Comment by hakenr** [comment]
@AJ1000 The new docs are now released.  
Check them out at https://havit.blazor.eu/getting-started.  
Any comments are welcome.  
(I got your feedback on the sidebar.)

**Comment by coderdnewbie** [comment]
Thanks, I tried it. One thing did not work. I get an Assets error when doing:

`<link href="@Assets["_content/Havit.Blazor.Components.Web.Bootstrap/bootstrap.min.css"]" rel="stylesheet" />`

I get the issue as I was using dotnet 8: CS0103: The name 'Assets' does not exist in the current context.
'Assets' is a dotnet 9.0 feature only

I had to do this instead for things to work:

```
<link rel="stylesheet" href="bootstrap/bootstrap.min.css" />
<link rel="stylesheet" href="_content/Havit.Blazor.Components.Web.Bootstrap/defaults.css" />

```
How are Assets working for you, is there a namespace?

The other thing I am not sure about is: When adding to the the Blazor Web app template, what Havit nuget packages go in the Main csproj file and what go in the .Client csproj file?

Thanks, the instructions look good, (By the way I am AJ1000 on my work machine and coderdnewbie on my home PC, I am the same person).

**Reply by hakenr** [reply]
You have to switch the setup to .NET 8 to get instructions for .NET 8 (without using the new `Assets`)
![image](https://github.com/user-attachments/assets/4ce4e4cb-8fe4-46a6-95f8-ce9dec5ec1ff)

**Comment by coderdnewbie** [comment]
Yes I see it now. Anyway I have started with dotnet 9.0 and made a new template and the upgrade instructions worked for me. Thanks

**Comment by MBNSoftware** [comment]
Hello,
I couldn't find another place to congratulate you on the new documentation so I'm doing it here. Very good job !!!

**Reply by hakenr** [reply]
@MBNSoftware Thanks for the kind words.  
That said, we are taking @AJ1000's suggestion very seriously and we're preparing a template for a new simple project (just a Blazor Web App with the Havit.Blazor library and `HxSidebar` - no authentication, no EF Core, no data layer, no API communication - a basic template for a quick start).  
With a bit of luck, it could be available by the end of this week.

**Comment by hakenr** [comment]
@AJ1000 
Simple project template created: https://github.com/havit/Havit.Blazor.SimpleBlazorWebAppTemplate
(Not linked from the website yet.)

Any suggestions before we announce this to the audience?

**Reply by AJ1000** [reply]
It looks perfect, for someone starting out, it is ideal to get started quickly. Thanks for doing this.

**Comment by AJ1000** [comment]
The template is fantastic
